internal/configsource/yamlconfigsource/README.md

```diff
@@ -0,0 +1,53 @@
+# YAML Config Source (Alpha)
+
+Use the YAML config source to inject [golang templates](https://pkg.go.dev/text/template)
+into the configuration.
+
+## Configuration
+
+Under the `config_sources:` use `yaml:` or `yaml/<name>:` to create a
+YAML config source.
+
+```yaml
+config_sources:
+  yaml:
+```
+
+The parameters of an YAML config source are passed to template to be processed.
+For example, assumint that `./templates/component_template` looks like:
+
+```terminal
+logs_path: {{ .glob_pattern }}
```

Comment:
My immediate impression with this example was that `glob_pattern` and `format` were keywords for the config source renderer and not generic.  Something like `my_globbed_path` and `my_var` may be more obvious imo.